internal/core/hls_index.html

```diff
@@ -33,18 +33,22 @@
 		});
 
 		hls.on(Hls.Events.ERROR, (evt, data) => {
-			if (data.fatal) {
+			if (data.type === Hls.ErrorTypes.MEDIA_ERROR)
+				hls.recoverMediaError();
+			else if (data.fatal) {
 				hls.destroy();
-
-				setTimeout(create, 2000);
+				setTimeout(() => create(video), 2000);
 			}
 		});
 
-		hls.loadSource('index.m3u8' + window.location.search);
+		hls.on(Hls.Events.MEDIA_ATTACHED, () => {
+			hls.loadSource('index.m3u8' + window.location.search);
+			hls.on(Hls.Events.MANIFEST_PARSED, () => {
+				video.play();
+			});
```

Comment:
move `hls.on(Hls.Events.MANIFEST_PARSED)` outside `hls.on(Hls.Events.MEDIA_ATTACHED)`, otherwise this hook will be setupped repeatedly every time the `MEDIA_ATTACHED` event is triggered.

Reply:
Yes, in theory this can happen. In practice, as you just once call "attachMedia", the callback on "MANIFEST_PARSED" will actually happen just once. Nevertheless, I moved it one level out now.